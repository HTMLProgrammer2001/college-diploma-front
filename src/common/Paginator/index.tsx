import React from 'react';
import {Pagination} from 'react-bootstrap';


type IPaginatorProps = {
	totalItems: number,
	curPage: number,
	size?: number,
	pageSize: number,
	setCur: (page: number) => void
};

const Paginator: React.FC<IPaginatorProps> = ({totalItems, curPage, setCur, pageSize, size = 5}) => {
	let hasPrev = curPage > size,
		totalPages = Math.ceil(totalItems/pageSize),
		hasNext = totalPages > (Math.floor((curPage - 1) / size) + 1) * size,
		curPaginator = Math.floor((curPage - 1) / size);

	if (totalPages <= 1)
		return null;

	return (
		<Pagination>
			{
				hasPrev &&
					<Pagination.Item onClick={() => setCur((curPaginator - 1) * size + 1)}>
						&lt;
					</Pagination.Item>
			}

			{
				new Array(size).fill('').map((val, index) => {
					const curItemNumber = curPaginator * size + index + 1;

					if(curItemNumber > totalPages)
						return null;

					return (
						<Pagination.Item
							key={index}
							active={curPage == curItemNumber}
							onClick={() => setCur(curItemNumber)}
						>
							{curItemNumber}
						</Pagination.Item>
					);
				})
			}

			{
				hasNext &&
					<Pagination.Item onClick={() => setCur((curPaginator + 1) * size + 1)}>
						&gt;
					</Pagination.Item>}
		</Pagination>
	);
};

export default Paginator;
