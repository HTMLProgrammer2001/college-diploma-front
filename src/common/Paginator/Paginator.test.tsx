import React from 'react';
import {shallow} from 'enzyme';

import Paginator from './';
import {Pagination} from 'react-bootstrap';


describe('Test paginator component', () => {
	it('Test render', () => {
		const wrapper = shallow(<Paginator totalItems={10} curPage={1} pageSize={5} setCur={null}/>);
		expect(wrapper.html()).toMatchSnapshot();
	});

	// it('Test', () => {
	// 	const curHandler = jest.fn(),
	// 		  wrapper = shallow(<Paginator totalItems={8} curPage={6} pageSize={5} setCur={curHandler}/>);
	//
	// 	//expect(wrapper.find(Pagination.Item).length).toBe(3);
	// 	expect(wrapper.find('.active').text()).toBe(6);
	// });
	//
	// it('Test next', () => {
	// 	expect(true).toBeTruthy();
	// });
	//
	// it('Test prev', () => {
	// 	expect(true).toBeTruthy();
	// });
});
