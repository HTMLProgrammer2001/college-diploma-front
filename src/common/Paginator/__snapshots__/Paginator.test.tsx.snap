// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test paginator component Test render 1`] = `"<ul class=\\"pagination\\"><li class=\\"page-item active\\"><span class=\\"page-link\\">1<span class=\\"sr-only\\">(current)</span></span></li><li class=\\"page-item\\"><a class=\\"page-link\\" role=\\"button\\" href=\\"#\\">2</a></li></ul>"`;
